### PR TITLE
Spinner's difficulty depends on beatmap's OD

### PR DESCRIPTION
**Pull request by EVAST9919**

### CODE REVIEW

File: osu.Game.Rulesets.Osu/Objects/Drawables/DrawableSpinner.cs [hunk 1]
```diff
@@ -22,6 +23,8 @@ public class DrawableSpinner : DrawableOsuHitObject
         private readonly Container circleContainer;
         private readonly DrawableHitCircle circle;
 
+        private float beatmapOd;
```
Comment (RemieRichards) [inline osu.Game.Rulesets.Osu/Objects/Drawables/DrawableSpinner.cs]: Overall Difficulty is frequently referred to as OD, should probably be maintained here too. (would require adding it as an abbreviation to the solution's DotSettings)

The ```//TODO``` you removed below also referred to it as OD.

File: osu.Game.Rulesets.Osu/Objects/Drawables/DrawableSpinner.cs [hunk 2]
```diff
@@ -71,6 +74,12 @@ public DrawableSpinner(Spinner s) : base(s)
             disc.Scale = scaleToCircle;
         }
 
+        [BackgroundDependencyLoader(permitNulls: true)]
+        private void load(OsuGame game)
+        {
+            beatmapOd = game?.Beatmap?.Value.Beatmap.BeatmapInfo.Difficulty.OverallDifficulty ?? 5;
```
Comment (smoogipooo) [inline osu.Game.Rulesets.Osu/Objects/Drawables/DrawableSpinner.cs]: You shouldn't be retrieving spins required like this. Override ApplyDefaults() in Spinner.cs and set a publicly visible local "SpinsRequired" or something, then refer to it in DrawableSpinner by `HitObject.SpinsRequired`. Look at DrumRoll.cs for an example of this.

**Comment by smoogipooo** [comment]
The correct algorithm is `Duration / 1000 * DifficultyRange(OD, 3, 5, 7.5)`.

File: osu.Game.Rulesets.Osu/Objects/Spinner.cs [hunk 3]
```diff
@@ -10,6 +12,18 @@ public class Spinner : OsuHitObject, IHasEndTime
...
+        /// </summary>
+        public double SpinsRequired { get; protected set; }
+
         public override bool NewCombo => true;
+
+        public override void ApplyDefaults(TimingInfo timing, BeatmapDifficulty difficulty)
+        {
+            base.ApplyDefaults(timing, difficulty);
+
+            SpinsRequired = Duration / 1000 * BeatmapDifficulty.DifficultyRange(difficulty.OverallDifficulty, 3, 5, 7.5);
```
Comment (smoogipooo) [inline osu.Game.Rulesets.Osu/Objects/Spinner.cs]: This should just be (int) casted.

**Review by smoogipoo — CHANGES_REQUESTED** [review]
Looks good for a merge after the change mentioned in my last comment is applied.

**Review by smoogipoo — DISMISSED** [review]

**Review by smoogipoo — APPROVED** [review]

**Comment by smoogipooo** [comment]
Good stuff.

**Pull request merged**